packages/components/nodes/multimodal/OpenAI/VLLMChain.ts

```diff
@@ -0,0 +1,201 @@
+import { OpenAI as OpenAIClient, ClientOptions } from 'openai'
+import { BaseChain, ChainInputs } from 'langchain/chains'
+import { ChainValues } from 'langchain/schema'
+import { BasePromptTemplate, ChatPromptTemplate, SystemMessagePromptTemplate } from 'langchain/prompts'
+import path from 'path'
+import { getUserHome } from '../../../src/utils'
+import fs from 'fs'
+
+/**
+ * Interface for the input parameters of the OpenAIVisionChain class.
+ */
+export interface OpenAIVisionChainInput extends ChainInputs {
+    openAIApiKey?: string
+    openAIOrganization?: string
+    throwError?: boolean
+    prompt?: BasePromptTemplate
+    configuration?: ClientOptions
+    imageUrls?: []
+    imageResolution?: string
+    temperature?: number
+    modelName?: string
+    maxTokens?: number
+    topP?: number
+    whisperConfig?: any
+}
+
+/**
+ * Class representing a chain for generating text from an image using the OpenAI
+ * Vision API. It extends the BaseChain class and implements the
+ * OpenAIVisionChainInput interface.
+ */
+export class VLLMChain extends BaseChain implements OpenAIVisionChainInput {
+    static lc_name() {
+        return 'VLLMChain'
+    }
+    prompt: BasePromptTemplate | undefined
+
+    inputKey = 'input'
+    outputKey = 'text'
+    imageUrls?: []
```

Comment:
we should change the name to something like `uploads` with interface.
This is being sent from the UI:
```
data: item.data,
type: item.type,
name: item.name,
mime: item.mime
```

Reply:
Updated.